kubernetes/structures_deployment.go

```
@@ -47,17 +61,28 @@ func expandDeploymentSpec(deployment []interface{}) (appsv1.DeploymentSpec, erro
 		obj.Strategy = expandDeploymentStrategy(v)
 	}
 
-	podSpec, err := expandPodSpec(in["template"].([]interface{}))
-	if err != nil {
-		return obj, err
-	}
+	obj.Template = expandPodTemplate(in["template"].([]interface{}))
+
+	return obj, nil
+}
 
-	obj.Template = corev1.PodTemplateSpec{
-		ObjectMeta: expandMetadata(in["template"].([]interface{})),
-		Spec:       podSpec,
+func expandPodTemplate(l []interface{}) corev1.PodTemplateSpec {
+	obj := corev1.PodTemplateSpec{}
+	if len(l) == 0 || l[0] == nil {
+		return obj
 	}
+	in := l[0].(map[string]interface{})
 
-	return obj, nil
+	obj.ObjectMeta = expandMetadata(in["metadata"].([]interface{}))
+
+	if v, ok := in["spec"].([]interface{}); ok && len(v) > 0 {
+		podSpec, err := expandPodSpec(in["spec"].([]interface{}))
+		if err != nil {
+			return obj
```

do we not want to bubble up this error?